### ISSUE TITLE
feat: Transformer LM using tensor ops

### ISSUE THREAD

**Issue by petrucristescu** [petrucristescu/churing#78]
## Summary

Build a character-level transformer language model end-to-end in pure Churing, using the tensor backend. Trains on a small text corpus and generates text autoregressively.

## New stdlib: `src/lib/attention.ch`

Scaled dot-product attention using `tensorMatMul`:

```
~attention Q,K,V (
    @d_k (toFloat (tensorCols Q))
    @scores (tensorMatMul Q (tensorTranspose K))
    @scaled (tensorScale (1.0 / sqrt d_k) scores)
    @weights (tensorSoftmaxRows scaled)
    tensorMatMul weights V)
```

## New stdlib: `src/lib/transformer.ch`

Single transformer block (self-attention + residual):

```
~transformerBlock x,Wq,Wk,Wv (
    @Q (tensorMatMul x Wq)
    @K (tensorMatMul x Wk)
    @V (tensorMatMul x Wv)
    @attn (attention Q K V)
    tensorAdd x attn)
```

## Example: `examples/lm/`

- `lm_train.ch` — tokenise a text corpus to char indices, train with next-char cross-entropy, save weights
- `lm_generate.ch` — load weights, take a seed string from `SEED` env var, generate N characters autoregressively
- `corpus.txt` — small training text (~500 chars)

## Model size

| Hyperparameter | Value |
|---|---|
| Vocabulary | ~30 chars (a-z + punctuation) |
| d_model | 16 |
| Sequence length | 8 |
| Attention heads | 1 |
| Layers | 1 |
| Parameters | ~3k |

Tiny enough to train in minutes on CPU tensors; benefits from GPU via `tensorToDevice`.

## Depends on

#75 (nn.ch with tensors), #76 (CUDA — optional, for GPU training)
